src/lib/sm.c

```diff
@@ -16,18 +17,46 @@ int sm_state(const struct sm *m)
 	return m->state;
 }
 
+static inline void sm_obs(const struct sm *m)
+{
+	struct timespec ts = {0};
+	struct tm tm;
+
+	clock_gettime(CLOCK_REALTIME, &ts);
+	gmtime_r(&ts.tv_sec, &tm);
+
+	tracef("%s pid: %lu sm_id: %lu %s |\n",
+		m->name, 111UL, m->id, m->conf[sm_state(m)].name);
+}
+
+void sm_to_sm_obs(const struct sm *from, const struct sm *to)
+{
+	struct timespec ts = {0};
+	struct tm tm;
+
+	clock_gettime(CLOCK_REALTIME, &ts);
+	gmtime_r(&ts.tv_sec, &tm);
+
+	tracef("%s-to-%s opid: %lu dpid: %lu id: %lu id: %lu |\n",
+		from->name, to->name, 111UL, 111UL, from->id, to->id);
+}
+
 void sm_init(struct sm *m,
 	     bool (*invariant)(const struct sm *, int),
 	     bool (*is_locked)(const struct sm *),
 	     const struct sm_conf *conf,
 	     int state)
 {
+	static sig_atomic_t id = 0;
```

Comment:
It's better to use uint64_t here. 
sig_atomic_t is uni32_t on some machines and can easily overflow.

Reply:
Changed.